src/services/auth.js

```diff
@@ -10,85 +10,104 @@ class Auth {
       responseType: 'token id_token',
       scope: 'openid profile email'
     })
-    this.session = JSON.parse(localStorage.getItem('auth'))
+
+    if (
+      !this.session &&
+      localStorage.auth &&
+      localStorage.userId &&
+      localStorage.userProfile
+    ) {
+      this.session = JSON.parse(localStorage.auth)
+      this.userProfile = JSON.parse(localStorage.userProfile)
+    }
   }
 
-  login (redirectURL) {
-    this.saveRedirectURL(redirectURL)
+  /* Action methods */
+  login (redirectTo) {
+    sessionStorage.after_login_redirect_url = redirectTo
     this.auth0.authorize()
   }
 
-  handleAuthentication (successCallback, errorCallback) {
+  logout () {
+    localStorage.removeItem('auth')
+    localStorage.removeItem('userId')
+    localStorage.removeItem('userProfile')
+    this.session = null
+    this.userId = null
+    this.userProfile = null
+    if (this.scheduledTimeout) {
+      clearTimeout(this.scheduledTimeout)
+    }
+  }
+
+  parseHash (hash) {
     return new Promise((resolve, reject) => {
-      this.auth0.parseHash(
-        {
-          hash: window.location.hash.split('?')[1]
-        },
-        (err, authResult) => {
-          if (authResult && authResult.accessToken && authResult.idToken) {
-            resolve(authResult)
-          } else if (err) {
-            alert(`Error: ${err.error}. Check the console for further details.`)
-            reject(err)
-          } else {
-            reject(new Error('Authentication did not work'))
-          }
+      this.auth0.parseHash({ hash }, (err, authResult) => {
+        if (authResult && authResult.accessToken && authResult.idToken) {
+          resolve(authResult)
+        } else if (err) {
+          alert(`Error: ${err.error}. Check the console for further details.`)
```

Comment:
Not quite the role of the authentication code to directly access the user visual space. Let the app (the caller) decide what to show the user.